### PR TITLE
Sync DUNS messaging on homepage with profile page messaging

### PR DESCRIPTION
**Pull request by jessieay**
* As an vendor, I want to see consistent status messages, if I see them at
  all, So, if the system is showing me status messages profile page and
  the homepage for the same thing, they should be consistent.

* Closes https://github.com/18F/micropurchase/issues/507

Example:

screen shot 2016-05-12 at 3 08 28 pm

### CODE REVIEW

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -72,27 +72,9 @@
...
-                    </li>
-                </ul>
-              </nav>
-              <!-- <span>Logged in as <%= current_user.name %></span> -->
-              <!-- <a class="usa-button usa-button-outline-inverse" href="<%= logout_path %>">Logout</a> -->
-              <!-- <a class="usa-button usa-button-outline-inverse" href="<%= edit_user_path(current_user) %>">Edit Profile</a> -->
+              <%= render partial: current_user.decorate.nav_drawer_partial, locals: { current_user: current_user } %>
             <% else %>
-              <a href="/login" class="button-login usa-button usa-button-outline-inverse">Login</a>
+              <%= render partial: Guest.new.decorate.nav_drawer_partial, locals: { current_user: current_user } %>
```
Comment (jessieay) [inline app/views/layouts/application.html.erb]: vision: eventually, we have no checks for `current_user` in our views and instead define methods on `Guest` that conform the the `User` API :)

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -1,22 +1,9 @@
...
-      flash.now[:success] = "Your DUNS number has been verified in Sam.gov"
-    elsif current_user.sam_rejected?
-      flash.now[:error] = "Your DUNS number was not found in Sam.gov. Please enter
-      a valid DUNS number to complete your profile. Check
-      https://www.sam.gov/sam/helpPage/SAM_Reg_Status_Help_Page.html to make
-      sure your DUNS number is correct. If you need any help email us at
-      micropurchase@gsa.gov"
-    end
+    @user = current_user
+    @user.decorate.sam_status_message_for(flash)
```
Comment (harrisj) [inline app/controllers/users_controller.rb]: Good move!

File: app/models/guest.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Guest
```
Comment (harrisj) [inline app/models/guest.rb]: So, is this to represent a non-logged in user?

File: app/models/sam_account_reckoner.rb [hunk 4]
```diff
@@ -1,28 +1,22 @@
 class SamAccountReckoner < Struct.new(:user)
-  def clear
-    if should_clear?
+  def set_default_sam_status
+    if should_clear_status?
       user.sam_status = :sam_pending
+    elsif user.duns_number.blank?
+      user.sam_status = :duns_blank
```
Comment (harrisj) [inline app/models/sam_account_reckoner.rb]: Good call on the duns_blank status

File: app/models/user.rb [hunk 5]
```diff
@@ -5,7 +5,11 @@ class User < ActiveRecord::Base
   validates :duns_number, duns_number: true
   validates :email, allow_blank: true, email: true
 
-  enum sam_status: { sam_pending: 0, sam_accepted: 1, sam_rejected: 2 }
+  enum sam_status: { duns_blank: 0, sam_accepted: 1, sam_rejected: 2, sam_pending: 3 }
```
Comment (harrisj) [inline app/models/user.rb]: Is there an issue on production renumbering the enum like this?

File: app/models/user.rb [hunk 6]
```diff
@@ -5,7 +5,11 @@ class User < ActiveRecord::Base
...
+
+  def decorate
```
Comment (harrisj) [inline app/models/user.rb]: I like this

File: app/view_models/auction_show_view_model.rb [hunk 7]
```diff
@@ -43,7 +43,7 @@ def auction_status_partial
   end
 
   def current_user_header_partial
-    current_user_has_no_sam_verification? ? "/components/no_sam_verification_header" : win_header_partial
+    win_header_partial
```
Comment (harrisj) [inline app/view_models/auction_show_view_model.rb]: Nice!

File: app/views/layouts/application.html.erb (hunk 1 above)
Comment (harrisj) [inline app/views/layouts/application.html.erb]: Nice!

**Comment by harrisj** [comment]
This looks great! Merging...

**Pull request merged**